Tests/AsyncQueueTests/ActorQueueTests.swift

```diff
@@ -39,47 +39,46 @@ final class ActorQueueTests: XCTestCase {
     func test_async_sendsEventsInOrder() async {
         let counter = Counter()
         for iteration in 1...1_000 {
-            systemUnderTest.async {
-                await counter.incrementAndExpectCount(equals: iteration)
+            systemUnderTest.async(on: counter) { counter in
+                counter.incrementAndExpectCount(equals: iteration)
             }
         }
-        await systemUnderTest.await { /* Drain the queue */ }
+        await systemUnderTest.await(on: counter) { _ in /* Drain the queue */ }
```

Comment:
I'm tempted to add a new `internal` method called `func allEnqueuedTasksSuspendedOrCompleted() async` to make this test call-site read more clearly since we have a few of these.